shared/utils/domains.test.ts

```diff
@@ -42,6 +42,13 @@ describe("#parseDomain", () => {
     });
   });
 
+  it("should return the same result when parsing the returned host", () => {
+    const customDomain = parseDomain("www.example.com");
+    const subDomain = parseDomain("myteam.example.com");
+    expect(parseDomain(customDomain.host)).toMatchObject(customDomain);
+    expect(parseDomain(subDomain.host)).toMatchObject(subDomain);
+  });
```

Comment:
The redirect logic relies on this behavior. It's always been true, just specifically testing for it now